src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/HelpersWindows.cs

```diff
@@ -532,23 +532,21 @@ private static unsafe void AddCryptAttribute(CryptographicAttributeObjectCollect
             collection.Add(new CryptographicAttributeObject(oid, attributeCollection));
         }
 
-        private static void GetParameters(CRYPT_ALGORITHM_IDENTIFIER cryptAlgorithmIdentifer, ref byte[] parameters)
+        private static void GetParameters(CRYPT_ALGORITHM_IDENTIFIER cryptAlgorithmIdentifer, int cbSize, ref byte[] parameters)
         {
-            if (cryptAlgorithmIdentifer.Parameters.cbData > 0)
+            if (cryptAlgorithmIdentifer.Parameters.cbData > 0 && cbSize > 0)
             {
-                CRYPT_RC2_CBC_PARAMETERS rc2Parameters;
-                unsafe
-                {
-                    int cbSize = 24;
-                    int cbSizeInit = cbSize;
-                    if (!CryptDecodeObject(CryptDecodeObjectStructType.X509_OCTET_STRING, cryptAlgorithmIdentifer.Parameters.pbData, (int)(cryptAlgorithmIdentifer.Parameters.cbData), &rc2Parameters, ref cbSize))
-                        throw Marshal.GetLastWin32Error().ToCryptographicException();
-                }
-
-                // Retrieve IV if available.
-                if (rc2Parameters.fIV > 0)
+                using (SafeHandle sh = Interop.Crypt32.CryptDecodeObjectToMemory(
+                    CryptDecodeObjectStructType.X509_OCTET_STRING,
+                    cryptAlgorithmIdentifer.Parameters.pbData,
+                    (int)cryptAlgorithmIdentifer.Parameters.cbData))
                 {
-                    parameters = rc2Parameters.ToByteArray();
+                    unsafe
+                    {
+                        DATA_BLOB* pDataBlob = (DATA_BLOB*)(sh.DangerousGetHandle());
+                        parameters = new byte[cbSize];
```

Comment:
`pDataBlob->cbData` says what the right answer is here, it should be used, or at least verified against the caller specified value.

Reply:
I used NET Framework as a source of truth. 
Some test sample like:
```
X509Store store = new X509Store("MY", StoreLocation.CurrentUser);
store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);

X509Certificate2Collection collection = (X509Certificate2Collection)store.Certificates;
X509Certificate2Collection fcollection = (X509Certificate2Collection)collection.Find(X509FindType.FindByTimeValid, DateTime.Now, false);

AlgorithmIdentifier algorithm = new AlgorithmIdentifier(new Oid("1.2.840.113549.3.7")); //3DES-CBC
ContentInfo contentInfo = new ContentInfo(new byte[] { 1, 2, 3 });
EnvelopedCms ecms = new EnvelopedCms(contentInfo, algorithm);

CmsRecipientCollection recipients = new CmsRecipientCollection(SubjectIdentifierType.IssuerAndSerialNumber, fcollection);
ecms.Encrypt(recipients);

byte[] encodedMessage = ecms.Encode();

ecms.Decode(encodedMessage);

Console.WriteLine(ecms.ContentEncryptionAlgorithm.Parameters.Length);
```
returns `16`.
The `pDataBlob->cbData` in CoreFX returns  `8`

How should I proceed according that?

Reply:
::facepalm::

Apparently NetFX isn't copying out the value of the parameters.  It's copying out the entire decoded CRYPTO_BLOB.  And therefore it's not the same size in a 32-bit and 64-bit process.  Huzzah.

That's not a bug that I think we should match, but it definitely tells me that at the end of the day essentially no one cares.

So, either
a) We make CoreFX return the right answer for the algorithms it returns any answer for.
b) Make the property exist, but CoreFX never sets it.
c) Make CoreFX return the objectively right answer (aka fix where it doesn't do it for AES)

At this point, I'll accept any of them, so long as CoreFX-Windows and CoreFX-Other agree.

Reply:
So I left the most clear option at the moment: 
> Make the property exist, but CoreFX never sets it.